tests/tests_test.go

```diff
@@ -0,0 +1,218 @@
+
+			if errors.IsAlreadyExists(err) {
+				log.Println(err)
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			_, err = sourceClient.AppsV1().Deployments("ocp-41583-longpvcname").Create(ctx, NewDeploymentFor41583(), metav1.CreateOptions{})
+			if errors.IsAlreadyExists(err) {
+				log.Println(err)
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			Eventually(func() v1.PodPhase {
+				podList, err := sourceClient.CoreV1().Pods("ocp-41583-longpvcname").List(ctx, metav1.ListOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				for _, p := range podList.Items {
+					if p.Status.Phase == "Running" {
+						podCmd := pods.PodCommand{
+							Pod:     &p,
+							RestCfg: sourceCfg,
+							Args:    []string{"sh", "-c", "dd if=/dev/zero of=/" + testFilePath + "file_1 bs=60M count=1"},
+						}
+						err = podCmd.Run()
+						if err != nil {
+							log.Println(err, "Failed running ls command inside destination Pod",
+								"pod", path.Join(p.Namespace, p.Name),
+								"command", "dd if=/dev/zero of=/"+testFilePath+"/file_1 bs=60M count=1;")
+						}
+					}
+					log.Println(fmt.Sprintf("Waiting for pod %s to be ready, currently in phase - %s", p.Name, p.Status.Phase))
+					return p.Status.Phase
+				}
+				return ""
+			}, time.Minute*5, time.Second).Should(Equal(v1.PodRunning))
+			log.Println("app deployed on the source cluster")
+		})
+
+		AfterEach(func() {
+			for _, ns := range namespaces {
+				err = sourceClient.CoreV1().Namespaces().Delete(ctx, ns, metav1.DeleteOptions{})
+				if err != nil {
+					log.Println(err)
+				}
+				err = hostClient.Delete(context.TODO(), &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: ns}})
+				if err != nil {
+					log.Println(err)
+				}
+			}
+		})
+
+		It("Should create a new migration", func() {
+			By("Creating a new migmigration")
+
+			migrationName := "foo"
+			migration := NewMigMigration(migrationName, migPlanName, false, false)
+			Expect(hostClient.Create(ctx, migration)).Should(Succeed())
+
+			fooMigration := &migapi.MigMigration{}
+
+			Eventually(func() string {
+				log.Println("waiting for dvm to complete")
+				dvms := &migapi.DirectVolumeMigrationList{}
+				Expect(hostClient.List(ctx, dvms, client.InNamespace(MigrationNamespace))).Should(Succeed())
+				for _, dvm := range dvms.Items {
+					if dvm.GenerateName == migrationName+"-" {
+						if dvm.Status.Itinerary == "VolumeMigration" {
+							dvmps := &migapi.DirectVolumeMigrationProgressList{}
+							Expect(hostClient.List(ctx, dvmps, client.MatchingLabels{"directvolumemigration": string(dvm.UID)})).Should(Succeed())
+							for _, dvmp := range dvmps.Items {
+								if dvmp.Status.TotalProgressPercentage == "100%" {
+									return dvm.Status.Phase
+								}
+							}
+						} else {
+							log.Println("Direct volume migration Failed")
```

Comment:
```
  name: foo-6rddt
  namespace: openshift-migration
  ownerReferences:
  - apiVersion: migration.openshift.io/v1alpha1
    controller: true
    kind: MigMigration
    name: foo
    uid: bc59b4ef-3ecb-4727-b70c-9767de9eaf76
  resourceVersion: "185152626"
  selfLink: /apis/migration.openshift.io/v1alpha1/namespaces/openshift-migration/directvolumemigrations/foo-6rddt
  uid: 8a5bcdfd-b4fd-4fcb-b638-eacf35658c88
spec:
  createDestinationNamespaces: true
  destMigClusterRef:
    name: host
    namespace: openshift-migration
  persistentVolumeClaims:
  - name: long-name-123456789011121314151617181920212223242526272829303132
    namespace: ocp-41583-longpvcname
    targetAccessModes:
    - ReadWriteOnce
    targetStorageClass: ocs-storagecluster-ceph-rbd
    verify: false
  srcMigClusterRef:
    name: e2e-object
    namespace: openshift-migration
status:
  observedDigest: ""
  phaseDescription: ""
```

When I tried the dvm is failing here. My assumption of what is happening:
1. migmigration controller creates DVM
2. the very first reconcile cycle, dvm phase  is ""
3. the test case catches that and declares it failure

Happened this twice. More logs:

```
2021/07/19 22:32:23 waiting for dvm to complete
2021/07/19 22:32:24 waiting for dvm to complete
2021/07/19 22:32:25 Direct volume migration Failed
STEP: Creating a new migmigration
• Failure [253.857 seconds]
Running migmigration when storage, cluster and plan are ready
/home/alpatel/go/src/github.com/konveyor/mig-controller/tests/tests_test.go:24
  Testing BZ #1965421, PVC name is more than 63 char long
  /home/alpatel/go/src/github.com/konveyor/mig-controller/tests/tests_test.go:54
    Should create a new migration [It]
    /home/alpatel/go/src/github.com/konveyor/mig-controller/tests/tests_test.go:121

    Expected
        <string>:
    to be equivalent to
        <string>: VolumeMigration

    /home/alpatel/go/src/github.com/konveyor/mig-controller/tests/tests_test.go:146
------------------------------
2021/07/19 22:32:25 Cleaning up is beginning


Summarizing 1 Failure:

[Fail] Running migmigration when storage, cluster and plan are ready Testing BZ #1965421, PVC name is more than 63 char long [It] Should create a new migration
/home/alpatel/go/src/github.com/konveyor/mig-controller/tests/tests_test.go:146

Ran 1 of 1 Specs in 273.447 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestMigmigration (273.45s)
FAIL

Ginkgo ran 1 suite in 4m35.45837171s
Test Suite Failed
make: *** [Makefile:132: e2e-test] Error 1
```

Reply:
convert this else into else if?

```
                                               } else if dvm.Status.Itinerary == "MigrationFailed" {
                                                        log.Println("Direct volume migration Failed")
                                                        Expect(dvm.Status.Itinerary).ShouldNot(BeEquivalentTo("MigrationFailed"))
                                               }
```